aiohttp/cookiejar.py

```diff
@@ -261,8 +261,43 @@ def filter_cookies(self, request_url: URL = URL()) -> "BaseCookie[str]":
                 request_origin = request_url.origin()
             is_not_secure = request_origin not in self._treat_as_secure_origin
 
+        # d: domain, d could be subdomain
+        # p: path
+        d = hostname
+
+        pairs = set()
+        while d:
+            p = request_url.path
+            while p:
+                pairs.add((d, p))
+                path_leftovers = p.rsplit("/", maxsplit=1)
+                # handle last element for rsplit
+                if len(path_leftovers) > 1:
+                    p = path_leftovers[0]
+                else:
+                    p = None
+
+            # handle last element for split
+            leftovers = d.split(".", maxsplit=1)
+            if len(leftovers) > 1:
+                d = leftovers[-1]
+            else:
+                d = None
+
+        # shared cookie, it should have max of 1 entry
+        pairs.add(("", "/"))
```

Comment:
Question from previous PR:
Maybe I'm just forgetting. But, is there anything in the code, or something in the spec which mentions this case?

```suggestion
        # shared cookie, it should have max of 1 entry
        pairs.add(("", "/"))
```

Reply:
Domain attribute without a value is undefined behavior, I added it based on the test cases data.

Reply:
Hmm, it does say:
> However, the user agent SHOULD ignore the cookie-av entirely.

But, I guess we'll just keep it compatible with the existing behaviour for now.